core/src/main/scala/com/databricks/labs/remorph/intermediate/plans.scala

```diff
@@ -138,14 +138,15 @@ abstract class Plan[PlanType <: Plan[PlanType]](_origin: Option[Origin] = Option
   }
 }
 
-abstract class LogicalPlan extends Plan[LogicalPlan] {
+abstract class LogicalPlan(_origin: Option[Origin] = Option.empty) extends Plan[LogicalPlan](_origin) {
 
   /**
    * Returns true if this expression and all its children have been resolved to a specific schema and false if it still
    * contains any unresolved placeholders. Implementations of LogicalPlan can override this (e.g.[[UnresolvedRelation]]
    * should return `false`).
    */
   lazy val resolved: Boolean = expressions.forall(_.resolved) && childrenResolved
+  var comments: Array[CommentNode] = Array()
```

Comment:
to be discussed, mutating a node at parse time is no big deal, alternately we could insert comment nodes before first child (or after depending on context), but that also requires mutating the tree